keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/ec2/ClusterSpec.kt

```diff
@@ -38,23 +33,26 @@ fun ClusterSpec.resolve(resolvedImages: ResolvedImages): Set<ServerGroup> =
   }
     .toSet()
 
-private fun ClusterSpec.resolveLaunchConfiguration(region: String, appVersion: String, imageId: String): LaunchConfiguration =
-  LaunchConfiguration(
-    appVersion = appVersion,
-    imageId = imageId,
-    instanceType = checkNotNull(overrides[region]?.launchConfiguration?.instanceType
+private fun ClusterSpec.resolveLaunchConfiguration(region: SubnetAwareRegionSpec): LaunchConfiguration {
+  val image = checkNotNull(overrides[region.region]?.launchConfiguration?.image
+    ?: defaults.launchConfiguration?.image) { "No image resolved / specified for ${region.region}" }
```

Comment:
Can the default for launch configuration image be null? Am I reading this right?

Reply:
Yes, because most specs will not want to specify an image explicitly. They'll get it from an artifact.

Reply:
Got it.